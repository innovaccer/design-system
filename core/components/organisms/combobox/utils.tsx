import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  inputTriggerRef?: any,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  multiSelect?: boolean,
  listRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef);
      break;
    case 'Enter':
      handleEnterKey(focusedOption, multiSelect, inputTriggerRef, listRef, setFocusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      inputTriggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (
  focusedOption: Element | undefined,
  multiSelect?: boolean,
  inputTriggerRef?: any,
  listRef?: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>
) => {
  (focusedOption as HTMLElement)?.click();
  if (!multiSelect) {
    inputTriggerRef.current.focus();
  } else {
    // to focus first option by default when last option is selected
    const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
    const listArr = Array.from(listItems);
    const index = listArr.findIndex((item) => {
      return item == focusedOption;
    });

    if (index === listArr.length - 1) {
      (listItems[0] as HTMLElement).focus();
      setFocusedOption && setFocusedOption(listItems[0]);
      listItems[0].scrollIntoView({ block: 'center' });
    }
  }
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any
) => {
  const listItems = listRef.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;
  }

  const targetOption = listItems[index];

  (targetOption as HTMLElement).focus();
  setFocusedOption && setFocusedOption(targetOption);
  targetOption?.scrollIntoView?.({ block: 'center' });
};
