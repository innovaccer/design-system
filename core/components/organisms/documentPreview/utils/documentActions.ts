/**
 * Utility functions for document actions in DocumentPreview
 * These functions provide working implementations for print and download functionality
 */

import { PreviewDocument } from '../types';

export interface DocumentActionOptions {
  filename?: string;
  autoShowPrintDialog?: boolean;
  printStyles?: string;
  fallbackBehavior?: 'open-new-tab' | 'show-error' | 'silent';
}

/**
 * Downloads a document (PDF or image) from the given URL
 * Works with both same-origin and cross-origin resources (with CORS)
 */
export const downloadDocument = async (
  document: PreviewDocument,
  options: DocumentActionOptions = {}
): Promise<void> => {
  const { filename, fallbackBehavior = 'open-new-tab' } = options;

  try {
    const downloadFilename = filename || generateFilename(document);

    let blob: Blob;

    if (document.src instanceof Blob) {
      blob = document.src;
    } else {
      const response = await fetch(document.src);

      if (!response.ok) {
        throw new Error(`HTTP ${response.status}: ${response.statusText}`);
      }

      blob = await response.blob();
    }
    const url = window.URL.createObjectURL(blob);

    const downloadLink = window.document.createElement('a');
    downloadLink.href = url;
    downloadLink.download = downloadFilename;
    downloadLink.style.display = 'none';

    window.document.body.appendChild(downloadLink);
    downloadLink.click();
    window.document.body.removeChild(downloadLink);

    setTimeout(() => {
      window.URL.revokeObjectURL(url);
    }, 100);
  } catch (error) {
    console.error('❌ Download failed:', error);

    const errorMessage = error instanceof Error ? error.message : 'Unknown error';

    switch (fallbackBehavior) {
      case 'open-new-tab':
        if (typeof document.src === 'string') {
          window.open(document.src, '_blank');
        }
        break;
      case 'show-error':
        alert(`Download failed: ${errorMessage}`);
        break;
      case 'silent':
        break;
    }
  }
};

/**
 * Prints a document (PDF or image)
 * For PDFs: Opens in new window with print dialog
 * For Images: Creates a print-optimized view
 */
export const printDocument = (document: PreviewDocument, options: DocumentActionOptions = {}): void => {
  const { autoShowPrintDialog = true, printStyles, fallbackBehavior = 'open-new-tab' } = options;

  try {
    const contentType = getDocumentContentType(document);

    if (contentType === 'pdf') {
      printPDF(document, autoShowPrintDialog);
    } else {
      printImage(document, { printStyles, autoShowPrintDialog });
    }
  } catch (error) {
    console.error('❌ Print failed:', error);

    if (fallbackBehavior === 'open-new-tab' && typeof document.src === 'string') {
      window.open(document.src, '_blank');
    }
  }
};

const printPDF = (document: PreviewDocument, autoShowPrintDialog: boolean): void => {
  if (!(typeof document.src === 'string')) {
    throw new Error('PDF printing requires a URL string, not a Blob');
  }

  const printWindow = window.open(document.src, '_blank', 'width=800,height=600');

  if (!printWindow) {
    throw new Error('Popup blocked - please allow popups for this site');
  }

  if (autoShowPrintDialog) {
    printWindow.onload = () => {
      printWindow.print();
    };
  }
};

const printImage = (
  document: PreviewDocument,
  options: { printStyles?: string; autoShowPrintDialog: boolean }
): void => {
  const { printStyles, autoShowPrintDialog } = options;

  const defaultStyles = `
    body {
      margin: 0;
      padding: 20px;
      text-align: center;
      font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif;
    }
    .document-title {
      font-size: 24px;
      font-weight: 600;
      margin-bottom: 20px;
      color: #333;
      page-break-after: avoid;
    }
    .document-subtitle {
      font-size: 14px;
      color: #666;
      margin-bottom: 30px;
      page-break-after: avoid;
    }
    .document-image {
      max-width: 100%;
      max-height: 90vh;
      height: auto;
      border: 1px solid #ddd;
      box-shadow: 0 2px 8px rgba(0,0,0,0.1);
    }
    @media print {
      body { padding: 0; }
      .document-title { font-size: 20px; margin-bottom: 15px; }
      .document-subtitle { font-size: 12px; margin-bottom: 20px; }
      .document-image { max-height: none; box-shadow: none; border: none; }
    }
  `;

  const finalStyles = printStyles || defaultStyles;

  const printWindow = window.open('', '_blank', 'width=800,height=600');

  if (!printWindow) {
    throw new Error('Popup blocked - please allow popups for this site');
  }

  const imageUrl = document.src instanceof Blob ? window.URL.createObjectURL(document.src) : document.src;

  const htmlContent = `
    <!DOCTYPE html>
    <html>
      <head>
        <title>Print - ${document.title}</title>
        <meta charset="utf-8">
        <style>${finalStyles}</style>
      </head>
      <body>
        <div class="document-title">${document.title}</div>
        ${document.subtitle ? `<div class="document-subtitle">${document.subtitle}</div>` : ''}
        <img 
          class="document-image" 
          src="${imageUrl}" 
          alt="${document.title}"
          onload="${autoShowPrintDialog ? 'window.print()' : ''}"
          onerror="document.body.innerHTML='<p>Error loading image for print</p>'"
        />
      </body>
    </html>
  `;

  printWindow.document.write(htmlContent);
  printWindow.document.close();

  if (document.src instanceof Blob) {
    setTimeout(() => {
      window.URL.revokeObjectURL(imageUrl);
    }, 5000);
  }
};

const generateFilename = (document: PreviewDocument): string => {
  const extension = getFileExtension(document);
  const sanitizedTitle = document.title
    .replace(/[^a-z0-9\s]/gi, '')
    .replace(/\s+/g, '_')
    .toLowerCase();

  return `${sanitizedTitle}${extension}`;
};

const getDocumentContentType = (document: PreviewDocument): string => {
  if (document.contentType) {
    return document.contentType;
  }

  const fileType = document.fileType?.toLowerCase();
  const srcLower = typeof document.src === 'string' ? document.src.toLowerCase() : '';

  if (fileType === 'pdf' || srcLower.includes('.pdf')) {
    return 'pdf';
  }

  if (
    fileType === 'image' ||
    ['jpg', 'jpeg', 'png', 'gif', 'webp', 'svg'].some((ext) => fileType === ext || srcLower.includes(`.${ext}`))
  ) {
    return 'image';
  }

  return 'auto';
};

const getFileExtension = (document: PreviewDocument): string => {
  const contentType = getDocumentContentType(document);

  switch (contentType) {
    case 'pdf':
      return '.pdf';
    case 'image': {
      const srcLower = typeof document.src === 'string' ? document.src.toLowerCase() : '';
      const fileType = document.fileType?.toLowerCase();

      if (fileType === 'png' || srcLower.includes('.png')) return '.png';
      if (fileType === 'gif' || srcLower.includes('.gif')) return '.gif';
      if (fileType === 'webp' || srcLower.includes('.webp')) return '.webp';
      if (fileType === 'svg' || srcLower.includes('.svg')) return '.svg';
      return '.jpg';
    }
    default:
      return '.file';
  }
};

export const downloadMultipleDocuments = async (
  documents: PreviewDocument[],
  options: DocumentActionOptions = {}
): Promise<void> => {
  const { fallbackBehavior = 'silent' } = options;

  for (let i = 0; i < documents.length; i++) {
    const doc = documents[i];

    await downloadDocument(doc, { ...options, fallbackBehavior });

    if (i < documents.length - 1) {
      await new Promise((resolve) => setTimeout(resolve, 500));
    }
  }
};

export const isDownloadSupported = (): boolean => {
  const a = window.document.createElement('a');
  return typeof a.download !== 'undefined';
};

export const isPrintSupported = (): boolean => {
  return typeof window.print === 'function';
};
