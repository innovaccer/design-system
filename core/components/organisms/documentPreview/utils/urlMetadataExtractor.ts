/**
 * Extracts metadata from URLs including S3 links
 */
export interface ExtractedMetadata {
  title: string;
  fileType: string;
  fileName: string;
  filePath?: string;
  encounterType?: string;
  contentType: 'image' | 'pdf' | 'txt' | 'xlsx' | 'xls' | 'tiff' | 'link' | 'auto';
}

export const extractMetadataFromUrl = (url: string): ExtractedMetadata => {
  try {
    const urlObj = new URL(url);
    const pathname = urlObj.pathname;

    const fileName = pathname.split('/').pop() || 'document';

    const fileExtension = fileName.includes('.') ? fileName.split('.').pop()?.toLowerCase() : undefined;

    const fileType = getFileTypeFromExtension(fileExtension);
    const contentType = getContentTypeFromExtension(fileExtension);

    const title = generateTitleFromFileName(fileName);

    const encounterType = determineEncounterType(url, pathname);

    return {
      title,
      fileType,
      fileName,
      filePath: pathname,
      encounterType,
      contentType,
    };
  } catch (error) {
    return {
      title: 'Document',
      fileType: 'unknown',
      fileName: 'document',
      contentType: 'auto',
    };
  }
};

const getFileTypeFromExtension = (extension?: string): string => {
  if (!extension) return 'unknown';

  const extensionMap: Record<string, string> = {
    pdf: 'pdf',
    jpg: 'image',
    jpeg: 'image',
    png: 'image',
    gif: 'image',
    webp: 'image',
    svg: 'image',
    txt: 'txt',
    xlsx: 'xlsx',
    xls: 'xls',
    tiff: 'tiff',
    tif: 'tiff',
    doc: 'doc',
    docx: 'docx',
  };

  return extensionMap[extension] || 'unknown';
};

const getContentTypeFromExtension = (
  extension?: string
): 'image' | 'pdf' | 'txt' | 'xlsx' | 'xls' | 'tiff' | 'link' | 'auto' => {
  if (!extension) return 'auto';

  const extensionMap: Record<string, 'image' | 'pdf' | 'txt' | 'xlsx' | 'xls' | 'tiff' | 'link' | 'auto'> = {
    pdf: 'pdf',
    jpg: 'image',
    jpeg: 'image',
    png: 'image',
    gif: 'image',
    webp: 'image',
    svg: 'image',
    txt: 'txt',
    xlsx: 'xlsx',
    xls: 'xls',
    tiff: 'tiff',
    tif: 'tiff',
  };

  return extensionMap[extension] || 'auto';
};

const generateTitleFromFileName = (fileName: string): string => {
  const nameWithoutExt = fileName.includes('.') ? fileName.substring(0, fileName.lastIndexOf('.')) : fileName;

  const formattedName = nameWithoutExt.replace(/[_-]/g, ' ').replace(/\s+/g, ' ').trim();

  return formattedName
    .split(' ')
    .map((word) => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
    .join(' ');
};

const determineEncounterType = (url: string, pathname: string): string => {
  const lowerUrl = url.toLowerCase();
  const lowerPath = pathname.toLowerCase();

  if (lowerUrl.includes('s3.amazonaws.com') || lowerUrl.includes('.s3.')) {
    if (lowerPath.includes('/medical/') || lowerPath.includes('/health/')) {
      return 'Medical Document';
    }
    if (lowerPath.includes('/legal/') || lowerPath.includes('/contracts/')) {
      return 'Legal Document';
    }
    if (lowerPath.includes('/financial/') || lowerPath.includes('/billing/')) {
      return 'Financial Document';
    }
    if (lowerPath.includes('/research/') || lowerPath.includes('/academic/')) {
      return 'Research Document';
    }
    return 'Business Document';
  }

  if (lowerPath.includes('/medical/') || lowerPath.includes('/health/')) {
    return 'Medical Document';
  }
  if (lowerPath.includes('/legal/') || lowerPath.includes('/contracts/')) {
    return 'Legal Document';
  }
  if (lowerPath.includes('/financial/') || lowerPath.includes('/billing/')) {
    return 'Financial Document';
  }
  if (lowerPath.includes('/research/') || lowerPath.includes('/academic/')) {
    return 'Research Document';
  }
  if (lowerPath.includes('/images/') || lowerPath.includes('/photos/')) {
    return 'Image Document';
  }

  const fileExt = pathname.split('.').pop()?.toLowerCase();
  if (fileExt === 'pdf') return 'PDF Document';
  if (['jpg', 'jpeg', 'png', 'gif'].includes(fileExt || '')) return 'Image Document';

  return 'Document';
};
