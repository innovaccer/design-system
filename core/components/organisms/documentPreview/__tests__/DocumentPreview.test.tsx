import * as React from 'react';
import { render } from '@testing-library/react';
import { DocumentPreview } from '@/index';
import { DocumentPreviewProps as Props } from '@/index.type';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const dimensions: Props['dimension'][] = ['regular', 'large'];

describe('DocumentPreview component', () => {
  const mapper = {
    dimension: valueHelper(dimensions, { required: true, iterate: true }),
    open: valueHelper([true], { required: true }),
    onClose: valueHelper([jest.fn()], { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<DocumentPreview {...attr} documentUrls={['https://example.com/test.pdf']} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);

  describe('DocumentPreview functionality', () => {
    const defaultProps: Props = {
      open: true,
      onClose: jest.fn(),
      documentUrls: ['https://example.com/test.pdf'],
      dimension: 'regular',
    };

    it('Should render with required props', () => {
      const { getByTestId } = render(<DocumentPreview {...defaultProps} />);
      expect(getByTestId('DocumentPreview')).toBeInTheDocument();
    });

    it('Should render with multiple documents', () => {
      const props = {
        ...defaultProps,
        documentUrls: [
          'https://example.com/test1.pdf',
          'https://example.com/test2.jpg',
          'https://example.com/test3.txt',
        ],
      };
      const { getByTestId } = render(<DocumentPreview {...props} />);
      expect(getByTestId('DocumentPreview')).toBeInTheDocument();
    });

    it('Should render with documents array', () => {
      const documents = [
        { src: 'https://example.com/test1.pdf', title: 'Test PDF' },
        { src: 'https://example.com/test2.jpg', title: 'Test Image' },
      ];
      const props = {
        ...defaultProps,
        documents,
        documentUrls: undefined,
      };
      const { getByTestId } = render(<DocumentPreview {...props} />);
      expect(getByTestId('DocumentPreview')).toBeInTheDocument();
    });

    it('Should render loading state', () => {
      const props = {
        ...defaultProps,
        loading: true,
      };
      const { getByTestId } = render(<DocumentPreview {...props} />);
      expect(getByTestId('DocumentPreview')).toBeInTheDocument();
    });

    it('Should render error state', () => {
      const props = {
        ...defaultProps,
        error: 'Failed to load document',
      };
      const { getByText } = render(<DocumentPreview {...props} />);
      expect(getByText('Failed to load document')).toBeInTheDocument();
    });

    it('Should render with disabled print', () => {
      const props = {
        ...defaultProps,
        disablePrint: true,
      };
      const { getByTestId } = render(<DocumentPreview {...props} />);
      expect(getByTestId('DocumentPreview')).toBeInTheDocument();
    });

    it('Should render with disabled download', () => {
      const props = {
        ...defaultProps,
        disableDownload: true,
      };
      const { getByTestId } = render(<DocumentPreview {...props} />);
      expect(getByTestId('DocumentPreview')).toBeInTheDocument();
    });

    it('Should render with navigation controls', () => {
      const props = {
        ...defaultProps,
        showNavigation: true,
        documentUrls: ['https://example.com/test1.pdf', 'https://example.com/test2.pdf'],
      };
      const { getByTestId } = render(<DocumentPreview {...props} />);
      expect(getByTestId('DocumentPreview')).toBeInTheDocument();
    });

    it('Should render with zoom controls', () => {
      const props = {
        ...defaultProps,
        showZoomControls: true,
      };
      const { getByTestId } = render(<DocumentPreview {...props} />);
      expect(getByTestId('DocumentPreview')).toBeInTheDocument();
    });

    it('Should render with expanded mode', () => {
      const props = {
        ...defaultProps,
        expandedMode: true,
      };
      const { getByTestId } = render(<DocumentPreview {...props} />);
      expect(getByTestId('DocumentDetailView')).toBeInTheDocument();
    });

    it('Should render with large dimension', () => {
      const props = {
        ...defaultProps,
        dimension: 'large' as const,
      };
      const { getByTestId } = render(<DocumentPreview {...props} />);
      expect(getByTestId('DocumentPreview')).toBeInTheDocument();
    });

    it('Should render with custom zoom settings', () => {
      const props = {
        ...defaultProps,
        initialZoom: 150,
        minZoom: 50,
        maxZoom: 300,
        onZoomChange: jest.fn(),
      };
      const { getByTestId } = render(<DocumentPreview {...props} />);
      expect(getByTestId('DocumentPreview')).toBeInTheDocument();
    });

    it('Should render with back navigation', () => {
      const props = {
        ...defaultProps,
        showBackNavigation: true,
      };
      const { getByTestId } = render(<DocumentPreview {...props} />);
      expect(getByTestId('DocumentPreview')).toBeInTheDocument();
    });

    it('Should render with program source component', () => {
      const ProgramSourceComponent = () => <div data-testid="program-source">Program Source</div>;
      const props = {
        ...defaultProps,
        programSourceComponent: ProgramSourceComponent,
      };
      const { getByTestId } = render(<DocumentPreview {...props} />);
      expect(getByTestId('DocumentPreview')).toBeInTheDocument();
    });
  });
});
