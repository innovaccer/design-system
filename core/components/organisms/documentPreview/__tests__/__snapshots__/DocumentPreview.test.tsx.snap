// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentPreview component 
  dimension: "large", open: [true], onClose: ["[Function]"]
 1`] = `
<body
  style="overflow: hidden;"
>
  <div
    class="Overlay-wrapper"
  >
    <div
      class="Row Overlay-container fade-in Overlay-container--open"
      data-layer="true"
      data-open="true"
      data-test="DesignSystem-SidesheetContainer"
      style="z-index: 1001;"
    >
      <div
        class="Col Col--6 Sidesheet Sidesheet--open Sidesheet-animation--open DocumentPreview-sidesheet Col--l-12"
        data-test="DocumentPreview"
      >
        <div
          class="Sidesheet-header Sidesheet-header--withSeperator"
        >
          <div
            class="Col"
            data-test="DesignSystem-Sidesheet--Header"
          >
            <div
              class="OverlayHeader"
              data-test="DesignSystem-OverlayHeader"
            >
              <div
                class="OverlayHeader-headerWrapper"
              >
                <h4
                  class="Heading Heading--m Heading--default Sidesheet-header--shiftLeft"
                  data-test="DesignSystem-OverlayHeader--heading"
                >
                  <div
                    class="ABC"
                    style="display: flex; justify-content: space-between;"
                  >
                    <div
                      class="d-flex"
                    >
                      
                      <div
                        class="PopperWrapper-trigger"
                      >
                        <h4
                          class="Heading Heading--m Heading--default DocumentPreview-title"
                          data-test="DesignSystem-Heading"
                        >
                          Test
                        </h4>
                      </div>
                    </div>
                    <div
                      class="d-flex"
                      style="gap: 8px;"
                    >
                      
                      <div
                        class="PopperWrapper-trigger"
                      >
                        <div>
                          <button
                            class="Button Button--regular Button--regularSquare Button--basic"
                            data-test="DocumentPreview--Print-Sidesheet"
                            disabled=""
                            tabindex="0"
                          >
                            <div
                              class="Button-icon"
                              data-test="DesignSystem-Button--Icon-Wrapper"
                            >
                              <i
                                class="material-symbols material-symbols-rounded Icon"
                                data-test="DesignSystem-Button--Icon"
                                role="button"
                                style="font-size: 16px; width: 16px;"
                              >
                                print
                              </i>
                            </div>
                          </button>
                        </div>
                      </div>
                      
                      <div
                        class="PopperWrapper-trigger"
                      >
                        <div>
                          <button
                            class="Button Button--regular Button--regularSquare Button--basic"
                            data-test="DocumentPreview--Download-Sidesheet"
                            disabled=""
                            tabindex="0"
                          >
                            <div
                              class="Button-icon"
                              data-test="DesignSystem-Button--Icon-Wrapper"
                            >
                              <i
                                class="material-symbols material-symbols-rounded Icon"
                                data-test="DesignSystem-Button--Icon"
                                role="button"
                                style="font-size: 16px; width: 16px;"
                              >
                                get_app
                              </i>
                            </div>
                          </button>
                        </div>
                      </div>
                      <button
                        class="Button Button--regular Button--regularSquare Button--basic"
                        data-test="DocumentPreview--Fullscreen"
                        tabindex="0"
                      >
                        <div
                          class="Button-icon"
                          data-test="DesignSystem-Button--Icon-Wrapper"
                        >
                          <i
                            class="material-symbols material-symbols-rounded Icon"
                            data-test="DesignSystem-Button--Icon"
                            role="button"
                            style="font-size: 16px; width: 16px;"
                          >
                            fullscreen
                          </i>
                        </div>
                      </button>
                    </div>
                  </div>
                  <div
                    class="d-flex"
                  >
                    <span
                      class="Text Text--subtle Text--regular"
                      data-test="DesignSystem-Text"
                    >
                      PDF Document • Document 1 of 1 • 2025-09-04
                    </span>
                  </div>
                </h4>
              </div>
            </div>
          </div>
          <div
            class="Col flex-grow-0"
            data-test="DesignSystem-Column"
          >
            
            <div
              class="PopperWrapper-trigger"
            >
              <button
                class="Button Button--regular Button--regularSquare Button--transparent"
                data-test="DesignSystem-Sidesheet--CloseButton"
                tabindex="0"
              >
                <div
                  class="Button-icon"
                  data-test="DesignSystem-Button--Icon-Wrapper"
                >
                  <i
                    class="material-symbols material-symbols-rounded Icon"
                    data-test="DesignSystem-Button--Icon"
                    role="button"
                    style="font-size: 20px; width: 20px;"
                  >
                    close
                  </i>
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          class="OverlayBody Sidesheet-body Sidesheet-body--firstPage"
          data-test="DesignSystem-Sidesheet--OverlayBody"
        >
          <div
            class="DocumentPreview"
            data-test="DocumentPreview--Content"
            id="filePreviewParent"
            style="position: relative; text-align: center; width: 100%; height: 100%; min-height: 85vh; display: inline-table; background-color: rgb(248, 249, 250);"
          >
            <div
              style="position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; overflow: auto;"
            >
              <div
                style="display: flex; flex-direction: column; align-items: center; justify-content: center; padding: 40px 20px; text-align: center; min-height: 200px;"
              >
                <span
                  class="Text Text--default Text--regular"
                  data-test="DesignSystem-Text"
                  style="margin-bottom: 16px;"
                >
                  Please download the file to view its contents.
                </span>
                <button
                  class="Button Button--regular Button--basic Button--iconAlign-left"
                  data-test="FilePreview--DownloadFallback"
                  tabindex="0"
                >
                  <div
                    class="Button-icon Button-icon--left Button-regularIcon--left"
                    data-test="DesignSystem-Button--Icon-Wrapper"
                  >
                    <i
                      class="material-symbols material-symbols-rounded Icon"
                      data-test="DesignSystem-Button--Icon"
                      role="button"
                      style="font-size: 16px; width: 16px;"
                    >
                      get_app
                    </i>
                  </div>
                  <span
                    class="Button-text"
                  >
                    Download
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div />
  <div
    class="Backdrop Backdrop--open Backdrop-animation--open"
    data-layer="true"
    data-opened="true"
    data-test="DesignSystem-Backdrop"
    style="z-index: 1000;"
  />
</body>
`;

exports[`DocumentPreview component 
  dimension: "regular", open: [true], onClose: ["[Function]"]
 1`] = `
<body
  style="overflow: hidden;"
>
  <div />
  <div
    class="Overlay-wrapper"
  >
    <div
      class="Row Overlay-container fade-in Overlay-container--open"
      data-layer="true"
      data-open="true"
      data-test="DesignSystem-SidesheetContainer"
      style="z-index: 1001;"
    >
      <div
        class="Col Col--6 Sidesheet Sidesheet--open Sidesheet-animation--open DocumentPreview-sidesheet Col--l-8"
        data-test="DocumentPreview"
      >
        <div
          class="Sidesheet-header Sidesheet-header--withSeperator"
        >
          <div
            class="Col"
            data-test="DesignSystem-Sidesheet--Header"
          >
            <div
              class="OverlayHeader"
              data-test="DesignSystem-OverlayHeader"
            >
              <div
                class="OverlayHeader-headerWrapper"
              >
                <h4
                  class="Heading Heading--m Heading--default Sidesheet-header--shiftLeft"
                  data-test="DesignSystem-OverlayHeader--heading"
                >
                  <div
                    class="ABC"
                    style="display: flex; justify-content: space-between;"
                  >
                    <div
                      class="d-flex"
                    >
                      
                      <div
                        class="PopperWrapper-trigger"
                      >
                        <h4
                          class="Heading Heading--m Heading--default DocumentPreview-title"
                          data-test="DesignSystem-Heading"
                        >
                          Test
                        </h4>
                      </div>
                    </div>
                    <div
                      class="d-flex"
                      style="gap: 8px;"
                    >
                      
                      <div
                        class="PopperWrapper-trigger"
                      >
                        <div>
                          <button
                            class="Button Button--regular Button--regularSquare Button--basic"
                            data-test="DocumentPreview--Print-Sidesheet"
                            disabled=""
                            tabindex="0"
                          >
                            <div
                              class="Button-icon"
                              data-test="DesignSystem-Button--Icon-Wrapper"
                            >
                              <i
                                class="material-symbols material-symbols-rounded Icon"
                                data-test="DesignSystem-Button--Icon"
                                role="button"
                                style="font-size: 16px; width: 16px;"
                              >
                                print
                              </i>
                            </div>
                          </button>
                        </div>
                      </div>
                      
                      <div
                        class="PopperWrapper-trigger"
                      >
                        <div>
                          <button
                            class="Button Button--regular Button--regularSquare Button--basic"
                            data-test="DocumentPreview--Download-Sidesheet"
                            disabled=""
                            tabindex="0"
                          >
                            <div
                              class="Button-icon"
                              data-test="DesignSystem-Button--Icon-Wrapper"
                            >
                              <i
                                class="material-symbols material-symbols-rounded Icon"
                                data-test="DesignSystem-Button--Icon"
                                role="button"
                                style="font-size: 16px; width: 16px;"
                              >
                                get_app
                              </i>
                            </div>
                          </button>
                        </div>
                      </div>
                      <button
                        class="Button Button--regular Button--regularSquare Button--basic"
                        data-test="DocumentPreview--Fullscreen"
                        tabindex="0"
                      >
                        <div
                          class="Button-icon"
                          data-test="DesignSystem-Button--Icon-Wrapper"
                        >
                          <i
                            class="material-symbols material-symbols-rounded Icon"
                            data-test="DesignSystem-Button--Icon"
                            role="button"
                            style="font-size: 16px; width: 16px;"
                          >
                            fullscreen
                          </i>
                        </div>
                      </button>
                    </div>
                  </div>
                  <div
                    class="d-flex"
                  >
                    <span
                      class="Text Text--subtle Text--regular"
                      data-test="DesignSystem-Text"
                    >
                      PDF Document • Document 1 of 1 • 2025-09-04
                    </span>
                  </div>
                </h4>
              </div>
            </div>
          </div>
          <div
            class="Col flex-grow-0"
            data-test="DesignSystem-Column"
          >
            
            <div
              class="PopperWrapper-trigger"
            >
              <button
                class="Button Button--regular Button--regularSquare Button--transparent"
                data-test="DesignSystem-Sidesheet--CloseButton"
                tabindex="0"
              >
                <div
                  class="Button-icon"
                  data-test="DesignSystem-Button--Icon-Wrapper"
                >
                  <i
                    class="material-symbols material-symbols-rounded Icon"
                    data-test="DesignSystem-Button--Icon"
                    role="button"
                    style="font-size: 20px; width: 20px;"
                  >
                    close
                  </i>
                </div>
              </button>
            </div>
          </div>
        </div>
        <div
          class="OverlayBody Sidesheet-body Sidesheet-body--firstPage"
          data-test="DesignSystem-Sidesheet--OverlayBody"
        >
          <div
            class="DocumentPreview"
            data-test="DocumentPreview--Content"
            id="filePreviewParent"
            style="position: relative; text-align: center; width: 100%; height: 100%; min-height: 85vh; display: inline-table; background-color: rgb(248, 249, 250);"
          >
            <div
              style="position: absolute; top: 0px; left: 0px; right: 0px; bottom: 0px; overflow: auto;"
            >
              <div
                style="display: flex; flex-direction: column; align-items: center; justify-content: center; padding: 40px 20px; text-align: center; min-height: 200px;"
              >
                <span
                  class="Text Text--default Text--regular"
                  data-test="DesignSystem-Text"
                  style="margin-bottom: 16px;"
                >
                  Please download the file to view its contents.
                </span>
                <button
                  class="Button Button--regular Button--basic Button--iconAlign-left"
                  data-test="FilePreview--DownloadFallback"
                  tabindex="0"
                >
                  <div
                    class="Button-icon Button-icon--left Button-regularIcon--left"
                    data-test="DesignSystem-Button--Icon-Wrapper"
                  >
                    <i
                      class="material-symbols material-symbols-rounded Icon"
                      data-test="DesignSystem-Button--Icon"
                      role="button"
                      style="font-size: 16px; width: 16px;"
                    >
                      get_app
                    </i>
                  </div>
                  <span
                    class="Button-text"
                  >
                    Download
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Backdrop Backdrop--open Backdrop-animation--open"
    data-layer="true"
    data-opened="true"
    data-test="DesignSystem-Backdrop"
    style="z-index: 1000;"
  />
</body>
`;
